### PR TITLE
Add NID column for multi-node generic webform exports.

### PR DESCRIPTION
**Pull request by torotil**

### CODE REVIEW

File: campaignion_csv.module [hunk 1]
```diff
@@ -114,3 +114,46 @@ function campaignion_csv_campaignion_csv_info() {
...
   return $export;
 }
+
+/**
+ * Implements hook_webform_results_download_submission_information_info_alter().
+ *
+ * Add a NID column right before the SID column.
+ */
+function campaignion_csv_webform_results_download_submission_information_info_alter(array &$info, $node = NULL, $options = NULL) {
+  if (!$options || !empty($options['multiple_nodes'])) {
```
Comment (katzmo) [inline campaignion_csv.module:124]: When are there no options and why do we want a nid column in this case?

File: campaignion_csv.module [hunk 2]
```diff
@@ -114,3 +114,46 @@ function campaignion_csv_campaignion_csv_info() {
...
+    $index = array_search('webform_sid', array_keys($info));
+    $columns['webform_nid'] = t('NID');
+    $info = array_slice($info, 0, $index) + $columns + array_slice($info, $index);
```
Comment (katzmo) [inline campaignion_csv.module:127]: Isn’t this the same as `array_splice($info, $index, 0, $columns)`?

File: campaignion_csv.module [hunk 3]
```diff
@@ -114,3 +114,46 @@ function campaignion_csv_campaignion_csv_info() {
...
+ * This hook is invoked if
+ * @link https://www.drupal.org/project/webform/issues/3086038 #3086038 @endlink
+ * is applied.
+ */
+function campaignion_csv_webform_results_download_submission_information_data_row($submission, array $options, $serial_start, $row_count) {
+  $data['webform_nid'] = $submission->nid;
+  return $data;
+}
+
+function _campaignion_csv_webform_results_download_submission_information_data($token, $submission, array $options, $serial_start, $row_count) {
```
Comment (katzmo) [inline campaignion_csv.module:161]: A docstring would be nice ;)

File: campaignion_csv.module (hunk 2 above)
Comment (torotil) [reply campaignion_csv.module]: No, this would ignore the array key. (I had that before.)

File: campaignion_csv.module (hunk 3 above)
Comment (torotil) [reply campaignion_csv.module]: Private functions don’t necessariliy need them. But, yes I can try to think of something.

File: campaignion_csv.module (hunk 1 above)
Comment (torotil) [reply campaignion_csv.module:124]: webform (without patch) doesn’t pass the `$node` and `$options`.

File: campaignion_csv.module (hunk 2 above)
Comment (torotil) [reply campaignion_csv.module:127]: That would remove the array key and use only the value instead.

File: campaignion_csv.module (hunk 3 above)
Comment (torotil) [reply campaignion_csv.module:161]: Private functions don’t necessarility need them. I can try to think of something useful though.

Comment (katzmo) [reply campaignion_csv.module:161]: I'm ok with "Helper function to add nid" or something, I just find it so much easier to see at a glance where a function ends and a new one begins when there is a docstring between them ;)

File: campaignion_csv.module (hunk 2 above)
Comment (katzmo) [reply campaignion_csv.module:127]: Even if you add the array [key => value] as the value? That's very confusing, but ok…

Comment (torotil) [reply campaignion_csv.module:127]:
```
php > $a = ['associative' => 'array'];
php > $b = [];
php > array_splice($b, 0, 0, $a);
php > var_dump($b);
array(1) {
  [0]=>
  string(5) "array"
}
```

Comment (katzmo) [reply campaignion_csv.module:127]: php is great!

File: campaignion_csv.module (hunk 3 above)
Comment (torotil) [reply campaignion_csv.module:161]: I’ve added a comment ;-)

Comment (katzmo) [reply campaignion_csv.module:161]: Thank you! :)

**Review by katzmo — APPROVED** [review]

**Pull request merged**